src/test/java/org/folio/rest/imp/InvoicesTest.java

```diff
@@ -1,114 +1,180 @@
+	private static final String ID = "id";
+	private static final String UUID = "8d3881f6-dd93-46f0-b29d-1c36bdb5c9f9";
+	private static final String EXIST_CONFIG_TENANT_LIMIT_10 = "test_diku_limit_10";
+	private static final String BAD_QUERY = "unprocessableQuery";
+	private static final String ID_FOR_INTERNAL_SERVER_ERROR = "168f8a86-d26c-406e-813f-c7527f241ac3";
+	private static final String TENANT_NAME = "diku";
+  private static final String QUERY_PARAM_NAME = "query";
+	private static final String EXISTING_VENDOR_INV_NO =  "existingVendorInvoiceNo";
+	private static final String VENDOR_INVOICE_NUMBER_FIELD = "vendorInvoiceNo";
 
-  @BeforeClass
-  public static void before() throws InterruptedException, ExecutionException, TimeoutException {
+	private static final int mockPort = NetworkUtils.nextFreePort();
+	private static final int okapiPort = NetworkUtils.nextFreePort();
 
-    // tests expect English error messages only, no Danish/German/...
-    Locale.setDefault(Locale.US);
+	private static Vertx vertx;
+	private static MockServer mockServer;
 
+	private static final Header EXIST_CONFIG_X_OKAPI_TENANT_LIMIT_10 = new Header(OKAPI_HEADER_TENANT, EXIST_CONFIG_TENANT_LIMIT_10);
+	private static final Header X_OKAPI_URL = new Header("X-Okapi-Url", "http://localhost:" + mockPort);
+	static final Header TENANT_HEADER = new Header(OKAPI_HEADER_TENANT, TENANT_NAME);
 
-    vertx = Vertx.vertx();
+	@BeforeClass
+	public static void setUpOnce(TestContext context) {
+		vertx = Vertx.vertx();
 
-    DeploymentOptions options = new DeploymentOptions();
+		mockServer = new MockServer(mockPort);
+		mockServer.start(context);
 
-    options.setConfig(new JsonObject().put("http.port", port).put(HttpClientMock2.MOCK_MODE, "true"));
-    options.setWorker(true);
+		RestAssured.baseURI = "http://localhost:" + okapiPort;
+		RestAssured.port = okapiPort;
+		RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
 
-    startVerticle(options);
+		final JsonObject conf = new JsonObject();
+		conf.put("http.port", okapiPort);
 
-  }
-
-  private static void startVerticle(DeploymentOptions options)
-    throws InterruptedException, ExecutionException, TimeoutException {
+		final DeploymentOptions opt = new DeploymentOptions().setConfig(conf);
+		vertx.deployVerticle(RestVerticle.class.getName(), opt, context.asyncAssertSuccess());
+	}
 
-    logger.info("Start verticle");
+	@AfterClass
+	public static void tearDownOnce(TestContext context) {
+		vertx.close(context.asyncAssertSuccess());
+		mockServer.close();
+	}
 
-    CompletableFuture<String> deploymentComplete = new CompletableFuture<>();
+	@Before
+	public void setUp() {
+		MockServer.serverRqRs.clear();
+	}
 
-    vertx.deployVerticle(RestVerticle.class.getName(), options, res -> {
-      if(res.succeeded()) {
-        deploymentComplete.complete(res.result());
-      }
-      else {
-        deploymentComplete.completeExceptionally(res.cause());
-      }
-    });
+  @Test
```

Comment:
I think it would be nice to move the Invoice tests to a separate class than the invoice lines, so that we can avoid the problems we had with Orders.

Reply:
We've added an extra point for splitting Unit tests to [MODINVOICE-20](https://issues.folio.org/browse/MODINVOICE-20). And I'm almost done with that.

Reply:
Ok. So, I will not change anything here then.

Reply:
Let me know if you are still these issues.